util/argo/argo.go

```diff
@@ -90,6 +109,21 @@ func FilterByName(apps []argoappv1.Application, name string) ([]argoappv1.Applic
 	return items, status.Errorf(codes.NotFound, "application '%s' not found", name)
 }
 
+// FilterAppSetsByName returns an applicationset
+func FilterAppSetsByName(appsets []argoappv1.ApplicationSet, name string) ([]argoappv1.ApplicationSet, error) {
```

Comment:
Looking at this function, I'm wondering why it returns a slice, when there can be either one or none results?

Reply:
I am using the same object newItems everywhere. First in `FilterAppSetsByName` and then for `FilterAppSetsByProjects`. `FilterAppSetsByName` would give one, but if q.name is empty, it would not enter that condition and directly move to `FilterAppSetsByProjects`

Reply:
https://github.com/argoproj/argo-cd/blob/23bb5a42d9f14c6d4df0fe3353fbf6b8d028ac3f/server/applicationset/applicationset.go#L120-L128

Reply:
Yeah, using a list here is convenient... but I think the List method shouldn't even accept a name "filter" (since the user should probably just use the Get endpoint).